crates/vm/src/arch/interpreter_preflight.rs

```diff
@@ -101,17 +101,13 @@ impl<F: Field, E> PreflightInterpretedInstance<F, E> {
         &self.inventory.executors
     }
 
-    pub fn filtered_execution_frequencies(&self) -> Vec<u32>
-    where
-        E: Send + Sync,
-    {
+    pub fn filtered_execution_frequencies(&self) -> Vec<u32> {
         let base_idx = get_pc_index(self.pc_base);
         self.pc_handler
             .par_iter()
-            .enumerate()
+            .zip_eq(&self.execution_frequencies)
```

Comment:
try `self.execution_frequencies.par_iter()`

Reply:
ah I think what you need is the above plus `p3_maybe_rayon::prelude::*;` but your fix is equivalent